helix-term/src/commands.rs

```diff
@@ -2637,6 +2637,34 @@ pub mod cmd {
         let (view, doc) = current!(cx.editor);
 
         view.ensure_cursor_in_view(doc, line);
+        Ok(())
+    }
+
+    fn setting(
+        cx: &mut compositor::Context,
+        args: &[Cow<str>],
+        _event: PromptEvent,
+    ) -> anyhow::Result<()> {
+        let runtime_config = &mut cx.editor.config;
+
+        if args.len() != 3 || args.get(1).map(|s| s.as_ref()) != Some("=") {
+            anyhow::bail!("Bad arguments. Usage: `:set key = field`");
+        }
+
+        match args[0].to_lowercase().as_ref() {
+            "scrolloff" => runtime_config.scrolloff = args[2].parse()?,
+            "scroll-lines" => runtime_config.scroll_lines = args[2].parse()?,
+            "mouse" => runtime_config.mouse = args[2].parse()?,
+            "line-number" => runtime_config.line_number = args[2].parse()?,
+            "middle-click_paste" => runtime_config.middle_click_paste = args[2].parse()?,
+            "smart-case" => runtime_config.smart_case = args[2].parse()?,
+            "auto-pairs" => runtime_config.auto_pairs = args[2].parse()?,
+            "auto-completion" => runtime_config.auto_completion = args[2].parse()?,
+            "completion-trigger-len" => runtime_config.completion_trigger_len = args[2].parse()?,
+            "auto-info" => runtime_config.auto_info = args[2].parse()?,
+            "true-color" => runtime_config.true_color = args[2].parse()?,
+            _ => anyhow::bail!("Unknown key `{}`.", args[0]),
```

Comment:
I am slightly worried about this since we can easily miss this.

Reply:
Well... That's why my two first versions were Serializing and Deserializing the original `Config` structure.
But then I understood you preferred a simple match: https://github.com/helix-editor/helix/pull/798#discussion_r730268931

Reply:
Sorry, I got confused for a bit. My original concern is that we are unnecessarily serializing and deserializing twice. And now my concern is we can easily miss the match.

I forgot why I say do a simple match back then. Not sure if there is a better way to do it without serializing and deserializing twice or doing a separate match (I mean separate as it the string is far from the original code).